### ISSUE TITLE
Elasticsearch 2.x does not allow `.` in field names

### ISSUE THREAD

**Issue by binarybana** [fabric8io/fluent-plugin-kubernetes_metadata_filter#14]
I tried using this plugin with ES 2.1 and get many errors like this:
```
[2016-01-28 18:23:58,459][DEBUG][action.bulk              ] [Gog] [logstash-2016.01.28][0] failed to execute bulk item (index) index {[logstash-2016.01.28][fluentd][AVKJekf4-Teb7njfaBNh], source[{"log":"2016/01/28 18:23:55 Client ip 10.244.65.1:52922 requesting /healthz probe servicing cmd nslookup kubernetes.default.svc.cluster.local localhost >/dev/null\n","stream":"stderr","time":"2016-01-28T18:23:55.459497426Z","docker":{"container_id":"c34fabafd21cf82441fd389cc71bf87a943be7ca7315ee6069307e7b95ab7e16"},"kubernetes":{"namespace_name":"kube-system","pod_id":"6a691072-c5de-11e5-831e-062fba24dd93","pod_name":"kube-dns-v9-v0kj9","container_name":"healthz","labels":{"k8s-app":"kube-dns","kubernetes.io/cluster-service":"true","version":"v9"},"host":"ip-10-2-63-0.us-west-2.compute.internal"},"tag":"kubernetes.var.log.containers.kube-dns-v9-v0kj9_kube-system_healthz-c34fabafd21cf82441fd389cc71bf87a943be7ca7315ee6069307e7b95ab7e16.log","@timestamp":"2016-01-28T18:23:55.459497426Z"}]}
MapperParsingException[Field name [kubernetes.io/cluster-service] cannot contain '.']
        at org.elasticsearch.index.mapper.object.ObjectMapper$TypeParser.parseProperties(ObjectMapper.java:278)
        at org.elasticsearch.index.mapper.object.ObjectMapper$TypeParser.parseObjectOrDocumentTypeProperties(ObjectMapper.java:223)
        at org.elasticsearch.index.mapper.object.ObjectMapper$TypeParser.parse(ObjectMapper.java:198)
        at org.elasticsearch.index.mapper.object.ObjectMapper$TypeParser.parseProperties(ObjectMapper.java:310)
        at org.elasticsearch.index.mapper.object.ObjectMapper$TypeParser.parseObjectOrDocumentTypeProperties(ObjectMapper.java:223)
        at org.elasticsearch.index.mapper.object.ObjectMapper$TypeParser.parse(ObjectMapper.java:198)
        at org.elasticsearch.index.mapper.object.ObjectMapper$TypeParser.parseProperties(ObjectMapper.java:310)
        at org.elasticsearch.index.mapper.object.ObjectMapper$TypeParser.parseObjectOrDocumentTypeProperties(ObjectMapper.java:223)
        at org.elasticsearch.index.mapper.object.RootObjectMapper$TypeParser.parse(RootObjectMapper.java:139)
        at org.elasticsearch.index.mapper.DocumentMapperParser.parse(DocumentMapperParser.java:140)
        at org.elasticsearch.index.mapper.DocumentMapperParser.parseCompressed(DocumentMapperParser.java:121)
        at org.elasticsearch.index.mapper.MapperService.parse(MapperService.java:391)
        at org.elasticsearch.cluster.metadata.MetaDataMappingService$2.execute(MetaDataMappingService.java:386)
        at org.elasticsearch.cluster.service.InternalClusterService$UpdateTask.run(InternalClusterService.java:388)
        at org.elasticsearch.common.util.concurrent.PrioritizedEsThreadPoolExecutor$TieBreakingPrioritizedRunnable.runAndClean(PrioritizedEsThreadPoolExecutor.java:231)
        at org.elasticsearch.common.util.concurrent.PrioritizedEsThreadPoolExecutor$TieBreakingPrioritizedRunnable.run(PrioritizedEsThreadPoolExecutor.java:194)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
        at java.lang.Thread.run(Thread.java:745)
```

Where the relevant line seems to be: `MapperParsingException[Field name [kubernetes.io/cluster-service] cannot contain '.']` It looks like the kubernetes official labels have the `kubernetes.io` prefix which makes ES unhappy due to [the `.` character being disallowed in ES 2.0](https://discuss.elastic.co/t/field-name-cannot-contain/33251)

Perhaps we should scrub these label names for `.`s and replace them with `_`?

**Comment by jimmidyson** [comment]
That is really annoying to say the least, but I understand the reasons why they've implemented it like that. Guess we'll need to de-dot as the link suggests - we can do that easily enough (PR welcome!).

I wonder if we should make nested objects for with same prefix, splitting on `/`?

**Comment by binarybana** [comment]
My knowledge of Ruby approaching 0 makes a PR a little out of my time budget at this moment unfortunately. :/ 

For the second one, I've been just accessing the logs through Kibana, where the nested fields don't seem to make much of a difference to the UI (AFAICT).

**Comment by jimmidyson** [comment]
No worries - I'll try to get a PR in in the next few days.

**Comment by binarybana** [comment]
Awesome, I'll gladly test it for you. :)

**Comment by jimmidyson** [comment]
Should be fixed by fcb6169c

**Comment by binarybana** [comment]
The fix works great. Thanks again!
